components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAOTest.java

```diff
@@ -0,0 +1,1690 @@
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+            addTestIdps(idPManagementDAO);
+            idPManagementDAO.deleteIdPs(tenantId);
+        }
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            String query = "SELECT * FROM IDP WHERE TENANT_ID=?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setInt(1, tenantId);
+            ResultSet resultSet = statement.executeQuery();
+            int resultSize = 0;
+            if (resultSet.next()) {
+                resultSize = resultSet.getRow();
+            }
+            statement.close();
+            assertEquals(resultSize, 0);
+        }
+    }
+
+    @DataProvider
+    public Object[][] deleteTenantRoleData() {
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, "Role1"},
+                {SAMPLE_TENANT_ID, "Role2"},
+        };
+    }
+
+    @Test(dataProvider = "deleteTenantRoleData")
+    public void testDeleteTenantRole(int tenantId, String role) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+            addTestIdps(idPManagementDAO);
+            idPManagementDAO.deleteTenantRole(tenantId, role, TENANT_DOMAIN);
+        }
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            String query = "SELECT * FROM IDP_ROLE WHERE TENANT_ID=? AND ROLE=?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setInt(1, tenantId);
+            statement.setString(2, role);
+            ResultSet resultSet = statement.executeQuery();
+            int resultSize = 0;
+            if (resultSet.next()) {
+                resultSize = resultSet.getRow();
+            }
+            statement.close();
+            assertEquals(resultSize, 0);
+        }
+    }
+
+    @Test(dataProvider = "deleteTenantRoleData")
+    public void testDeleteTenantRoleException(int tenantId, String role) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        IdPManagementDAO idPManagementDAO = new IdPManagementDAO();
+
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+            addTestIdps(idPManagementDAO);
+        }
+        assertThrows(IdentityProviderManagementException.class, () ->
+                idPManagementDAO.deleteTenantRole(tenantId, role, TENANT_DOMAIN));
```

Comment:
Add inside the try block.

Reply:
For having a exception this block should placed out the try block.